tcc: add page (#9894)
tldr-pages/tldr: pages/common/tcc.md

# tcc

> > A tiny C compiler that can run C source files as scripts and otherwise has command line options similar to gcc.

Reviewer: ```suggestion
> A tiny C compiler that can run C source files as scripts and otherwise has command line options similar to `gcc`.
```

> More information: <https://bellard.org/tcc/tcc-doc.html>.

> - Compile and link `a.c` and `b.c` to generate the executable `myprog`:

Reviewer: ```suggestion
- Compile and link `a.c` and `b.c` to generate an executable:
```
How about this?

> - Compile and link `file1.c` and `file2.c` to generate an executable:

Reviewer 2: ```suggestion
- Compile and link 2 source files to generate an executable:
```

> `tcc -o myprog a.c b.c`

Reviewer: ```suggestion
`tcc -o {{executable_name}} {{path/to/a.c}} {{path/to/b.c}}`
```
Note: `{{}}` used here is used for highlighting placeholders in our clients.

> `tcc -o {{executable_name}} {{path/to/a.c}} {{path/to/b.c}}`

Reviewer 3: Converter TlDr -> CLIP (Command Line Interface Pages) can't recognize any of `{{path/to/a.c}} {{path/to/b.c}}` placeholders. To fix this issue add `_file` suffix right before .c extension.

Note: issue was fixed manually [here](https://github.com/command-line-interface-pages/cli-pages/pull/47).

Reviewer: Or it can be replace with `file1` and `file2` instead of `a` and `b`.

Reviewer 3: Sure, or with `{{path/to/source_file.c}}`.

Author: I've now implemented those suggestions.

> - Directly run `a.c` like a script and pass `args` to it:

Reviewer: ```suggestion
- Directly run `a.c` like a script and pass arguments to it:
```

> `tcc -run a.c args`

Reviewer: ```suggestion
`tcc -run {{path/to/a.c}} {{arguments}}`
```

> - Directly run `source_file.c` like a script and pass arguments to it:

Reviewer 2: ```suggestion
- Directly run an input file like a script and pass arguments to it:
```

- Interpret C source files with a shebang inside the file:

> `#!/full/path/to/tcc -run`

Reviewer 2: This doesn't match the TLDR syntax.

Author: The alternative is to remove this part for not being a terminal command, but surely that's worse than leaving it in as valuable information. I can remove it if you guys really want me to, but I feel that that would be needlessly prohibitive without providing any actual benefit. Especially since this is a useful and unique differentiating feature of `tcc`

Reviewer 2: Ok, sure, we will leave it.
